pkg/ccl/changefeedccl/checkpoint/checkpoint_test.go

```diff
@@ -164,6 +166,142 @@ func TestCheckpointMake(t *testing.T) {
 	}
 }
 
+func TestCheckpointRestore(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	ts := func(wt int64) hlc.Timestamp {
+		return hlc.Timestamp{WallTime: wt}
+	}
+
+	for name, tc := range map[string]struct {
+		initialHighWater       hlc.Timestamp
+		oldCheckpointTs        hlc.Timestamp
+		oldCheckpointSpans     []roachpb.Span
+		newCheckpoint          jobspb.TimestampSpansMap
+		expectedError          error
+		expectedFrontierForOld checkpointSpans
+		expectedFrontierForNew checkpointSpans
+	}{
+		"old checkpoint ts is empty": {
+			initialHighWater: ts(0),
+			oldCheckpointTs:  ts(0),
+			oldCheckpointSpans: []roachpb.Span{
+				{Key: roachpb.Key("a"), EndKey: roachpb.Key("b")},
+			},
+			newCheckpoint: jobspb.TimestampSpansMap{
+				Entries: []jobspb.TimestampSpansMap_Entry{
+					{
+						Timestamp: ts(2),
+						Spans:     []roachpb.Span{{Key: roachpb.Key("b"), EndKey: roachpb.Key("c")}},
+					},
+				},
+			},
+			expectedError: errors.New("checkpoint timestamp is empty"),
+		},
+		"new checkpoint ts is empty": {
+			initialHighWater: ts(0),
+			oldCheckpointTs:  ts(2),
+			oldCheckpointSpans: []roachpb.Span{
+				{Key: roachpb.Key("a"), EndKey: roachpb.Key("b")},
+			},
+			newCheckpoint: *jobspb.NewTimestampSpansMap(map[hlc.Timestamp]roachpb.Spans{
```

Comment:
nit: maybe you could change the field type to be a pointer so you don't need to dereference it here nor get the address of it later

Reply:
Done.